### ISSUE TITLE
how to install plugins ?

### ISSUE THREAD

**Issue by qtwrk** [firehol/netdata#2145]
Hi guys , this is amazing stuff , but I got little problem about how to install plugin.

and please forgive my stupidity , I google'd around but couldn't find anything really useful.

https://github.com/firehol/netdata/wiki/Add-more-charts-to-netdata

I found this page and there are lots of plugins 
but how am I supposed to install them ?

for example the first plugin 
apache python v2 or v3.

should I grab that "python.d.plugin" and place it into /usr/libexec/netdata/plugins.d ?
what about "apache.chart.py" and "python.d/apache.conf" ?
where should I place them into ?

I also noticed there a lot of files in /usr/libexec/netdata/, what am I supposed to do with them ?

Could someone please explain to me a little how to install it ?
that'd be much appreciated :)

**Comment by ktsaou** [comment]
Hi,

everything in that page is already installed. Most plugins support auto-detection too, so if the application you want to monitor allows local processes to monitor it, netdata plugins will just work.

In many of the cases though, you need to configure the application you want to monitor, to allow netdata connect to it.

So, for apache, you can find the netdata configuration at `/etc/netdata/python.d/apache.conf`. This file says that netdata tries these URLs:

```
localhost:
  name : 'local'
  url  : 'http://localhost/server-status?auto'

localipv4:
  name : 'local'
  url  : 'http://127.0.0.1/server-status?auto'

localipv6:
  name : 'local'
  url  : 'http://::1/server-status?auto'
```

So, if your apache is listeninig on port 80, and you have mod_status enabled, the plugin will work.

You can verify your apache is exposing the proper information by running::

```
curl "URL"
```

where `URL` is any of the URLs above.

You can run the plugin by hand to see the error it gets, with this:

```sh
# become user netdata
su -s /bin/bash netdata

# run python plugin for apache
/usr/libexec/netdata/plugins.d/python.d.plugin debug 1 apache
```

**Comment by ktsaou** [comment]
Cleaning up. I am closing this now. If you need help, just post.

**Comment by cocodrino** [comment]
hi @ktsaou sorry if it's not ok ask in a closed issue but I've a related question, you said that we just need configurate the required file, for instance /etc/netdata/python.d/apache.conf ...now I'm interested in the portcheck plugin but this route is empty /etc/netdata/python.d/ 

do I need create a file portcheck.conf and automagically the plugin will be activate?...I read about that you're migrating from python plugins to go plugins, so must I activate the go.d plugin portcheck instead?

thank you so much...netdata is amazing! ..

**Comment by ilyam8** [comment]
@cocodrino 	

see https://docs.netdata.cloud/docs/configuration-guide/#configuration-guide

tl;dr
 - `cd` to `/etc/netdata`
 - `./edit-config python.d/portcheck.conf`
 - modify;exit (config will be saved to `/etc/netdata/python.d/` on exit)

next time go directly to `/etc/netdata/python.d/`

**Comment by cocodrino** [comment]
great...thanks @ilyam8
